scripts/globals/porter_moogle_util.lua

```diff
@@ -0,0 +1,337 @@
+end
+
+----------------------------------------------------------------------
+-- desc : Begins the storage or retrieval process based on the items
+--        supplied in the trade.
+----------------------------------------------------------------------
+function doTrade(player, trade, e)
+    local slipId, slipCount = getSlipId(trade);
+    if (slipId == 0) then return; end;
+    
+    local storableItemIds = getStorableItems(player, trade, slipId);
+    if (slipCount > 1 and #storableItemIds == 0) then
+        return;
+    end
+        
+    if (#storableItemIds > 0) then
+        storeItems(player, storableItemIds, slipId, e);
+    else
+        startRetrieveProcess(player, e.RETRIEVE_EVENT_ID, slipId);
+    end 
+end
+
+----------------------------------------------------------------------
+-- desc : Retrieves the selected item from storage, removes it from
+--        the slip's extra data, displays a message to the user, and
+--        updates the user's event data.
+----------------------------------------------------------------------
+function retrieveItem(player, option, RETRIEVE_DIALOG_ID, ITEM_CANNOT_BE_OBTAINED)
+    local slipId = player:getLocalVar('slipId');
+    if (slipId ~= 0 and slipId ~= nil) then
+        local extra = player:getRetrievableItemsForSlip(slipId);
+        local itemsOnSlip = getItemsOnSlip(extra, slipId);
+        local retrievedItemId = itemsOnSlip[option + 1];
+        
+        if (player:hasItem(retrievedItemId) or player:getFreeSlotsCount() == 0) then
+            player:messageSpecial(ITEM_CANNOT_BE_OBTAINED, retrievedItemId);
+        else
+            local k = find(slipItems[slipId], retrievedItemId);
+            local extraId = math.floor((k - 1) / 8);
+            local bitmask = extra[extraId + 1];
+            if bitmask < 0 then
+                bitmask = bitmask + 256;
+            end
+
+            bitmask = clearBit(bitmask, bit((k - 1) % 8 + 1));
+            extra[extraId + 1] = bitmask;
+
+            player:retrieveItemFromSlip(slipId, retrievedItemId, extraId, bitmask);
+            player:messageSpecial(RETRIEVE_DIALOG_ID, retrievedItemId, nil, nil, retrievedItemId, false);
+        end
+        
+        local params = int8ToInt32(extra);
+        player:updateEvent(params[1], params[2], params[3], params[4], params[5], params[6], slipId);
+    end 
+end
+
+----------------------------------------------------------------------
+-- desc : Completes the event.
+----------------------------------------------------------------------
+function buyStorageSlip(player, option, ITEM_CANNOT_BE_OBTAINED, ITEM_OBTAINED, NOT_HAVE_ENOUGH_GIL)
+    -- This is just because hilarious.
+    option = math.floor(option / 16) + (option % 16);
+    local hasItem = player:hasItem(slipIds[option]);
+    if (hasItem or player:getFreeSlotsCount() == 0) then
+        player:messageSpecial(ITEM_CANNOT_BE_OBTAINED, slipIds[option]);
+        return;
+    end
+
+    if (player:getGil() < 1000) then
```

Comment:
i *think* this does pretty much the same as the if block below, and the message can probably be moved in an else there instead